src/user/identity.ts

```diff
@@ -0,0 +1,73 @@
+import * as base58 from 'base58-universal/main.js'
+import eccOperations from 'keystore-idb/ecc/operations'
+import rsaOperations from 'keystore-idb/rsa/operations'
+import utils from 'keystore-idb/utils'
+
+import { base64UrlEncode, isRSAKeystore } from '../common'
+import { getKeystore } from '../keystore'
+
+
+const ED_DID_PREFIX: ArrayBuffer = new Uint8Array([ 0xed, 0x01 ]).buffer
+const RSA_DID_PREFIX: ArrayBuffer = new Uint8Array([ 0x00, 0xf5, 0x02 ]).buffer
+
+
+/**
+ * Create a DID key to authenticate with and wrap it in a JWT.
+ */
+export const didJWT = async () => {
+  const ks = await getKeystore()
+  const isRSA = isRSAKeystore(ks)
+
+  // Parts
+  const header = {
+    alg: isRSA ? 'RS256' : 'Ed25519',
+    typ: 'JWT'
+  }
+
+  const payload = {
+    iss: await didKey(),
+    exp: Math.floor((Date.now() + 5 * 60 * 1000) / 1000), // JWT expires in 5 minutes
+  }
+
+  // Encode parts in JSON & Base64Url
+  const encodedHeader = base64UrlEncode(JSON.stringify(header))
+  const encodedPayload = base64UrlEncode(JSON.stringify(payload))
+
+  // Signature
+  const operator = isRSA
+    ? rsaOperations
+    : eccOperations
+
+  const signed = await operator.signBytes(
```

Comment:
This is all fine, but it's essentially what `keystore-idb` does under the hood.

Which means there's a short coming with the `keystore` pkg :sweat_smile: 

I actually ran into a similar issue as well.... I'll create an issue over on that repo to add byte operations to the keystore interface as well so you don't have to do all this stuff

Reply:
https://github.com/fission-suite/keystore-idb/issues/22

Reply:
wait actually misread this a bit on first pass. we are dealing with strings :sweat_smile: 
so
```ts

  // Signature
  const operator = isRSA
    ? rsaOperations
    : eccOperations

  const signed = await operator.signBytes(
    utils.strToArrBuf(`${encodedHeader}.${encodedPayload}`, 8),
    ks.writeKey.privateKey,
    ks.cfg.hashAlg
  )

  const encodedSignature = base64UrlEncode(
    utils.arrBufToStr(signed, 8)
  )
```

can actually just be shortened to 
```ts
  const sig = await ks.sign(`${encodedHeader}.${encodedPayload}`)
  const encodedSiganture = makeBase64UrlSafe(sig)
```

Unless I'm missing something here :thinking:

Reply:
Yeah, we should definitely change it in keystore-idb. I needed `signBytes` originally for something different, I needed the array buffer for usage with the crypto api. But it turned out that was unnecessary, so ya.

Anyhow, long story short, you're right, I can use `ks.sign`, __except__, I need to be able to pass a custom `charSize`. Here it needs to be `8`, not `16` (what it is currently set to).

What do you think? Should I add that as an option to the `sign` method, or am I doing something wrong?

Reply:
you can pass a custom `charSize` in the `keystore-idb` cfg. But it might be the case that you want your keystore to default to `16` for instance, but `8` for a single function. We could add an optional `cfg` param to every method in the keystore for overriding the cfg on method call. Thoughts?

Reply:
I guess we should add it as an option.
Seeing that all the javascript stuff will need to be `utf-16`,
and other, less common, things could be `utf-8`.

Reply:
Okay I went over to have a look at it, started messing around, and just figured that I'd finish it off :sweat_smile: 

Check out the PR here: we can get it merged & deployed

Reply:
oops: https://github.com/fission-suite/keystore-idb/pull/24

Reply:
Done 👍